lib/fetch/util.js

```diff
@@ -1174,6 +1176,39 @@ function buildContentRange (rangeStart, rangeEnd, fullLength) {
   return contentRange
 }
 
+// A Stream, which pipes the response to zlib.createInflate() or
+// zlib.createInflateRaw() depending on the first byte of the Buffer.
+// If the lower byte of the first byte is 0x08, then the stream is
+// interpreted as a zlib stream, otherwise it's interpreted as a
+// raw deflate stream.
+class InflateStream extends Transform {
+  _transform (chunk, encoding, callback) {
+    if (!this._inflateStream) {
+      this._inflateStream = (chunk[0] & 0x0F) === 0x08
```

Comment:
@mcollina 
@ronag 
I would like to add this to the code, just to be sure. wdyt?

```suggestion
      if (chunk.length === 0) {
        callback()
        return
      }
      this._inflateStream = (chunk[0] & 0x0F) === 0x08
```

Reply:
Doesn't hurt